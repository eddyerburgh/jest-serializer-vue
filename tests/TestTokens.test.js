const helpers = require('./helpers.js');

import { shallowMount } from '@vue/test-utils';
import TestTokens from './components/TestTokens.vue';

describe('TestTokens.vue', () => {
  test('Remove data-server-rendered, data-test, data-testid, data-test-id by default', () => {
    helpers.mockSettings({});

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-server-rendered removed', () => {
    helpers.mockSettings({
      removeServerRendered: true,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: false,
      removeDataCy: false,
      removeClassTest: false,
      removeIdTest: false
    });

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-test removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: true,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: false,
      removeDataCy: false,
      removeClassTest: false,
      removeIdTest: false
    });

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-testid removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: true,
      removeDataTestId: false,
      removeDataQa: false,
      removeDataCy: false,
      removeClassTest: false,
      removeIdTest: false
    });

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-test-id removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: true,
      removeDataQa: false,
      removeDataCy: false,
      removeClassTest: false,
      removeIdTest: false
    });

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-qa removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: true,
      removeDataCy: false,
      removeClassTest: false,
      removeIdTest: false
    });

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only data-cy removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: false,
      removeDataCy: true,
      removeClassTest: false,
      removeIdTest: false
    });

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only id="test" removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: false,
      removeDataCy: false,
      removeClassTest: false,
      removeIdTest: true
    });

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Only class="test" removed', () => {
    helpers.mockSettings({
      removeServerRendered: false,
      removeDataTest: false,
      removeDataTestid: false,
      removeDataTestId: false,
      removeDataQa: false,
      removeDataCy: false,
      removeClassTest: true,
      removeIdTest: false
    });

    const wrapper = shallowMount(TestTokens);

    expect(wrapper)
      .toMatchSnapshot();
  });
});
