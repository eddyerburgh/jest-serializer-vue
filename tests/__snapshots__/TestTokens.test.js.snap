// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestTokens.vue Only class="test" removed 1`] = `
<ul id="tokens" class="tokens token">
  <li data-server-rendered="true">
    data-server-rendered="true"
  </li>
  <li data-test="data-test">
    data-test="token"
  </li>
  <li data-testid="data-testid">
    data-testid="token"
  </li>
  <li data-test-id="data-test-id">
    data-test-id="token"
  </li>
  <li data-qa="data-qa">
    data-qa="token"
  </li>
  <li data-cy="data-cy">
    data-cy="token"
  </li>
  <li id="test">
    id="test"
  </li>
  <li id="testId">
    id="testId"
  </li>
  <li id="test-id">
    id="test-id"
  </li>
  <li>
    class="test"
  </li>
  <li>
    class="testToken"
  </li>
  <li>
    class="test-token"
  </li>
  <li>
    class="test-token testToken test"
  </li>
  <li class="active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;

exports[`TestTokens.vue Only data-cy removed 1`] = `
<ul id="tokens" class="tokens token">
  <li data-server-rendered="true">
    data-server-rendered="true"
  </li>
  <li data-test="data-test">
    data-test="token"
  </li>
  <li data-testid="data-testid">
    data-testid="token"
  </li>
  <li data-test-id="data-test-id">
    data-test-id="token"
  </li>
  <li data-qa="data-qa">
    data-qa="token"
  </li>
  <li>
    data-cy="token"
  </li>
  <li id="test">
    id="test"
  </li>
  <li id="testId">
    id="testId"
  </li>
  <li id="test-id">
    id="test-id"
  </li>
  <li class="test">
    class="test"
  </li>
  <li class="testToken">
    class="testToken"
  </li>
  <li class="test-token">
    class="test-token"
  </li>
  <li class="test-token testToken test">
    class="test-token testToken test"
  </li>
  <li class="test active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;

exports[`TestTokens.vue Only data-qa removed 1`] = `
<ul id="tokens" class="tokens token">
  <li data-server-rendered="true">
    data-server-rendered="true"
  </li>
  <li data-test="data-test">
    data-test="token"
  </li>
  <li data-testid="data-testid">
    data-testid="token"
  </li>
  <li data-test-id="data-test-id">
    data-test-id="token"
  </li>
  <li>
    data-qa="token"
  </li>
  <li data-cy="data-cy">
    data-cy="token"
  </li>
  <li id="test">
    id="test"
  </li>
  <li id="testId">
    id="testId"
  </li>
  <li id="test-id">
    id="test-id"
  </li>
  <li class="test">
    class="test"
  </li>
  <li class="testToken">
    class="testToken"
  </li>
  <li class="test-token">
    class="test-token"
  </li>
  <li class="test-token testToken test">
    class="test-token testToken test"
  </li>
  <li class="test active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;

exports[`TestTokens.vue Only data-server-rendered removed 1`] = `
<ul id="tokens" class="tokens token">
  <li>
    data-server-rendered="true"
  </li>
  <li data-test="data-test">
    data-test="token"
  </li>
  <li data-testid="data-testid">
    data-testid="token"
  </li>
  <li data-test-id="data-test-id">
    data-test-id="token"
  </li>
  <li data-qa="data-qa">
    data-qa="token"
  </li>
  <li data-cy="data-cy">
    data-cy="token"
  </li>
  <li id="test">
    id="test"
  </li>
  <li id="testId">
    id="testId"
  </li>
  <li id="test-id">
    id="test-id"
  </li>
  <li class="test">
    class="test"
  </li>
  <li class="testToken">
    class="testToken"
  </li>
  <li class="test-token">
    class="test-token"
  </li>
  <li class="test-token testToken test">
    class="test-token testToken test"
  </li>
  <li class="test active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;

exports[`TestTokens.vue Only data-test removed 1`] = `
<ul id="tokens" class="tokens token">
  <li data-server-rendered="true">
    data-server-rendered="true"
  </li>
  <li>
    data-test="token"
  </li>
  <li data-testid="data-testid">
    data-testid="token"
  </li>
  <li data-test-id="data-test-id">
    data-test-id="token"
  </li>
  <li data-qa="data-qa">
    data-qa="token"
  </li>
  <li data-cy="data-cy">
    data-cy="token"
  </li>
  <li id="test">
    id="test"
  </li>
  <li id="testId">
    id="testId"
  </li>
  <li id="test-id">
    id="test-id"
  </li>
  <li class="test">
    class="test"
  </li>
  <li class="testToken">
    class="testToken"
  </li>
  <li class="test-token">
    class="test-token"
  </li>
  <li class="test-token testToken test">
    class="test-token testToken test"
  </li>
  <li class="test active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;

exports[`TestTokens.vue Only data-test-id removed 1`] = `
<ul id="tokens" class="tokens token">
  <li data-server-rendered="true">
    data-server-rendered="true"
  </li>
  <li data-test="data-test">
    data-test="token"
  </li>
  <li data-testid="data-testid">
    data-testid="token"
  </li>
  <li>
    data-test-id="token"
  </li>
  <li data-qa="data-qa">
    data-qa="token"
  </li>
  <li data-cy="data-cy">
    data-cy="token"
  </li>
  <li id="test">
    id="test"
  </li>
  <li id="testId">
    id="testId"
  </li>
  <li id="test-id">
    id="test-id"
  </li>
  <li class="test">
    class="test"
  </li>
  <li class="testToken">
    class="testToken"
  </li>
  <li class="test-token">
    class="test-token"
  </li>
  <li class="test-token testToken test">
    class="test-token testToken test"
  </li>
  <li class="test active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;

exports[`TestTokens.vue Only data-testid removed 1`] = `
<ul id="tokens" class="tokens token">
  <li data-server-rendered="true">
    data-server-rendered="true"
  </li>
  <li data-test="data-test">
    data-test="token"
  </li>
  <li>
    data-testid="token"
  </li>
  <li data-test-id="data-test-id">
    data-test-id="token"
  </li>
  <li data-qa="data-qa">
    data-qa="token"
  </li>
  <li data-cy="data-cy">
    data-cy="token"
  </li>
  <li id="test">
    id="test"
  </li>
  <li id="testId">
    id="testId"
  </li>
  <li id="test-id">
    id="test-id"
  </li>
  <li class="test">
    class="test"
  </li>
  <li class="testToken">
    class="testToken"
  </li>
  <li class="test-token">
    class="test-token"
  </li>
  <li class="test-token testToken test">
    class="test-token testToken test"
  </li>
  <li class="test active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;

exports[`TestTokens.vue Only id="test" removed 1`] = `
<ul id="tokens" class="tokens token">
  <li data-server-rendered="true">
    data-server-rendered="true"
  </li>
  <li data-test="data-test">
    data-test="token"
  </li>
  <li data-testid="data-testid">
    data-testid="token"
  </li>
  <li data-test-id="data-test-id">
    data-test-id="token"
  </li>
  <li data-qa="data-qa">
    data-qa="token"
  </li>
  <li data-cy="data-cy">
    data-cy="token"
  </li>
  <li>
    id="test"
  </li>
  <li>
    id="testId"
  </li>
  <li>
    id="test-id"
  </li>
  <li class="test">
    class="test"
  </li>
  <li class="testToken">
    class="testToken"
  </li>
  <li class="test-token">
    class="test-token"
  </li>
  <li class="test-token testToken test">
    class="test-token testToken test"
  </li>
  <li class="test active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;

exports[`TestTokens.vue Remove data-server-rendered, data-test, data-testid, data-test-id by default 1`] = `
<ul id="tokens" class="tokens token">
  <li>
    data-server-rendered="true"
  </li>
  <li>
    data-test="token"
  </li>
  <li>
    data-testid="token"
  </li>
  <li>
    data-test-id="token"
  </li>
  <li data-qa="data-qa">
    data-qa="token"
  </li>
  <li data-cy="data-cy">
    data-cy="token"
  </li>
  <li id="test">
    id="test"
  </li>
  <li id="testId">
    id="testId"
  </li>
  <li id="test-id">
    id="test-id"
  </li>
  <li class="test">
    class="test"
  </li>
  <li class="testToken">
    class="testToken"
  </li>
  <li class="test-token">
    class="test-token"
  </li>
  <li class="test-token testToken test">
    class="test-token testToken test"
  </li>
  <li class="test active">
    class="test active"
  </li>
  <li class>
    class=""
  </li>
</ul>
`;
